api/src/main/java/jakarta/persistence/ConnectionConsumer.java

```diff
@@ -16,25 +16,26 @@
 
 package jakarta.persistence;
 
-import java.sql.Connection;
-import java.sql.SQLException;
-
 /**
- * An executable action which makes use of a {@linkplain Connection JDBC connection}.
+ * An executable action which makes use of a native database connection.
+ * The connection is usually a JDBC connection.
+ *
+ * @param <C> the connection type, usually {@code java.sql.Connection}
  *
  * @see ConnectionFunction
  * @see EntityManager#runWithConnection(ConnectionConsumer)
  *
  * @since 3.2
  */
 @FunctionalInterface
-public interface ConnectionConsumer {
+public interface ConnectionConsumer<C> {
 	/**
 	 * Execute the action using the given connection.
 	 *
 	 * @param connection the connection to use
 	 *
-	 * @throws SQLException if a problem occurs calling JDBC
+	 * @throws Exception if a problem occurs calling the connection,
+	 *                   usually a {@code java.sql.SQLException}
 	 */
-	void accept(Connection connection) throws SQLException;
+	void accept(C connection) throws Exception;
```

Comment:
`<C, E extends Exception>` with `accept(C connection) throws E;` - not sure if that isn't too much.. The benefit here would be that the client has a possibility to explicitly cover particular exception type instead of having to deal with everything. WDYT?

Reply:
ummmmmm well I was assuming that the `EM` would just catch any checked exception and rethrow it as a `PersistenceException`

But also: won't `E` just get inferred to whatever type the lambda throws, which may as well be `Exception` *anyway*? It's essentially unconstrained...

Reply:
grr, I've used bad example on my end :-/